crates/proto/src/rr/record_type.rs

```diff
@@ -149,6 +149,7 @@ impl FromStr for RecordType {
     /// ```
     fn from_str(str: &str) -> ProtoResult<Self> {
         // TODO missing stuff?
+        debug_assert!(str.chars().all(|x| char::is_digit(x, 36)));
```

Comment:
Here, by using the `is_digit` function, with a base of 36, we can match any a-z, A-Z, or 0-9 character. The digits are the reason we cannot use `char::is_uppercase`; `NSEC3` records would fail, as `3` is not an uppercase character.

Reply:
ah, yeah. it's really just a helpful debug statement anyway.